### ISSUE TITLE
Trap should be allowed to use a different detection map

### ISSUE THREAD

**Issue by gijzelaerr** [transientskp/tkp#391]
In the TraP the source finding step should be allowed to use a detection map different from the analysis map. The detection map should be the non-primary beam corrected output of the awimager.
This feature already exists for PySE but it is not usable within the TraP.

@jdswinbank: Dario, can you give us a use-case for this? Is it not covered by the 'monitoringlist' functionality?

**Comment by ajstewart** [comment]
I just wanted to add that I think this would be a very nice feature to include. Sources are much cleaner to extract from the non-pb corrected images, and this is a common method used to extract sources from images. I would argue for it to be above a low priority.

**Comment by timstaley** [comment]
Yes, it sounds like this is pretty useful. @gijzelaerr , initially I was confused over this too, but it's all tucked away under the 'sourcefinding' functionality (so no impact on the database code). Basically, for every observation we feed the sourcefinder two images, one primary-beam corrected and one not. The sourcefinder performs island-detection on the uncorrected image (flat noise-profile makes this easier) but then runs Gaussian fitting on the primary-beam corrected image (so correct flux values). Apparently this is already implemented in Pyse, but we'd need to figure out how to pair up images and pass them as a list to the sourcefinder appropriately.

**Comment by AntoniaR** [comment]
Not quite sure why this has been assigned to me, unless you were after more discussion...

There is a slight difference what people are asking for, although in theory we could have both.

1. A single deep "detection" image can be fed into TraP at the start of the pipeline run. This is a normal image. This is how I (and I think a few others) frequently use TraP, but we don't want the flux measurements to be included in the lightcurves and variability parameters. So it's sort of like we're populating the monitoring list with an image - but the biggest advantage is being able to check new sources detected against the deep catalogue, something that the monitoring list is not currently used for. Being able to have this is a reasonably high priority for me (but lower than e.g. quality control checks for non-LOFAR images).

2. TraP uses two different images for each time step, the uncorrected image and the primary beam corrected image (as described). To be completely honest, I cannot see me using this functionality in the near future and I think it would be superseded by using a deep image at the start of the pipeline run. So personally, I would say this is very low priority. Perhaps @dariocarbone6 and @ajstewart could comment as they requested this functionality?

**Comment by gijzelaerr** [comment]
you marked it as high priority on you wiki page :) But not sure if you have seen it, you know assign multiple people to the same issue.

**Comment by AntoniaR** [comment]
This is an ongoing wish list item but not essential for R7. Moving to the long term milestone.

**Comment by AntoniaR** [comment]
Basically the issue was that you could use a different image for detecting the sources and then extract the fluxes in the image you want to process. So it is still using PySE, but implementing the `[--detection-image DETECTION_IMAGE]` which is in the standalone PySE.
However, the use case was from people who no longer work with TraP and this is a low priority in my opinion/
